main.cu

```diff
@@ -0,0 +1,277 @@
+#include "common.h"
+#include "mmio.h"
+#include "test.h"
+
+float clockElapsed(cudaEvent_t evt_start, cudaEvent_t evt_stop) {
+    cudaEventSynchronize(evt_stop);
+
+    float elapsedTime = 0;
+
+    cudaEventElapsedTime(&elapsedTime, evt_start, evt_stop);
+    elapsedTime *= 1000;  // Returns in microseconds
+
+    return elapsedTime;
+}
+
+int main(int argc, char** argv) {
+    // report precision of floating-point
```

Comment:
There is so much logic here. Even though this is the main method, you could modularize this into methods and functions.